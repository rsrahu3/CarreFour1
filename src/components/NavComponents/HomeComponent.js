import * as React from 'react'
import Header from './../common/Header'
import { connect } from 'react-redux'
import thunk from './../../middleware/thunk'

class HomeComponent extends React.Component {
  componentDidMount () {
    this.props.showData()
  }

  render () {
    return <div id="headerDiv" >
      <Header active="home"/>
      <table className="table">
        <thead>
          <tr>
            <th>ID</th>
            <th>NAME</th>
            <th>SALARY</th>
            <th>AGE</th>
          </tr>
        </thead>
        <tbody>
          {this.props.data.map((item) => {
            return (<tr key={item.id}>

              {Object.keys(item).map(element => {
                return (<td key={element}>{item[element]}</td>)
              })}

            </tr>)
          })}
        </tbody>
      </table>
          This is Home Component
    </div>
  }
}

const mapStateToProps = (state) => {
  return {
    data: state.data
  }
}
const mapDispatchToProps = (dispatch) => {
  return {
    showData: () => dispatch(thunk({ url: 'http://dummy.restapiexample.com/api/v1/employees', type: 'populate_data' }))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(HomeComponent)
