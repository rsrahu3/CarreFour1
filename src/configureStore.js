import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage'; // defaults to localStorage for web
import homeReducer from './reducers/homeReducer';
import todoReducer from './reducers/todoReducer';
import { combineReducers } from 'redux';
import { createStore } from 'redux';
const rootReducer = combineReducers({ home: homeReducer, toDo: todoReducer });

const persistConfig = {
  key: 'root',
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export default () => {
  let store = createStore(persistedReducer);
  let persistor = persistStore(store);
  return { store, persistor };
};
