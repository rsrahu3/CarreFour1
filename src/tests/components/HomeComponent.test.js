import React from 'react';
import {shallow,configure} from 'enzyme';
import HomeComponent from './../../components/NavComponents/HomeComponent';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store'
configure({ adapter: new Adapter() });
import toJson from 'enzyme-to-json';

const initialState = {
data : [{"id":"142591","employee_name":"test65","employee_salary":"123","employee_age":"23","profile_image":""},
                {"id":"142593","employee_name":"mz\/x","employee_salary":"123","employee_age":"23","profile_image":""},
                {"id":"142594","employee_name":"MMODD66ttt","employee_salary":"80","employee_age":"22","profile_image":""},
                {"id":"142603","employee_name":"bew","employee_salary":"200000","employee_age":"25","profile_image":""}]



}; // here it is possible to pass in any middleware if needed into //configureStore
const mockStore = configureStore();
let wrapper;
let store;
beforeEach(() => {  //creates the store with any initial state or middleware needed  
  store = mockStore(initialState)
 })




describe("Home Component Screen test",()=>{
    it('form check',()=>{
       wrapper = shallow(<HomeComponent store={store}/>)
         expect(toJson(wrapper).children[0].node.props.data.length).toBe(4);
         expect(toJson(wrapper)).toMatchSnapshot();
    })
    
})