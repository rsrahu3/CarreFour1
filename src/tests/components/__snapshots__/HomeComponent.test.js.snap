// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Screen test form check 1`] = `
<ContextProvider
  value={null}
>
  <HomeComponent
    data={
      Array [
        Object {
          "employee_age": "23",
          "employee_name": "test65",
          "employee_salary": "123",
          "id": "142591",
          "profile_image": "",
        },
        Object {
          "employee_age": "23",
          "employee_name": "mz/x",
          "employee_salary": "123",
          "id": "142593",
          "profile_image": "",
        },
        Object {
          "employee_age": "22",
          "employee_name": "MMODD66ttt",
          "employee_salary": "80",
          "id": "142594",
          "profile_image": "",
        },
        Object {
          "employee_age": "25",
          "employee_name": "bew",
          "employee_salary": "200000",
          "id": "142603",
          "profile_image": "",
        },
      ]
    }
    showData={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
